vantage/virtual_tag_config_resource.go

```diff
@@ -53,13 +58,47 @@ func (r VirtualTagConfigResource) Schema(ctx context.Context, req resource.Schem
 				Computed: true,
 				NestedObject: schema.NestedAttributeObject{
 					Attributes: map[string]schema.Attribute{
-						"filter": schema.StringAttribute{
+						"business_metric_token": schema.StringAttribute{
 							Optional:            true,
-							Computed:            true,
+							Description:         "The token of the associated BusinessMetric.",
+							MarkdownDescription: "The token of the associated BusinessMetric.",
+						},
+						"cost_metric": schema.SingleNestedAttribute{
+							Attributes: map[string]schema.Attribute{
+								"aggregation": schema.SingleNestedAttribute{
+									Attributes: map[string]schema.Attribute{
+										"tag": schema.StringAttribute{
+											Required:            true,
+											Description:         "The tag to aggregate on.",
+											MarkdownDescription: "The tag to aggregate on.",
+										},
+									},
+									CustomType: resource_virtual_tag_config.AggregationType{
+										ObjectType: types.ObjectType{
+											AttrTypes: resource_virtual_tag_config.AggregationValue{}.AttributeTypes(ctx),
+										},
+									},
+									Required: true,
+								},
+								"filter": schema.StringAttribute{
+									Required:            true,
+									Description:         "The filter VQL for the cost metric.",
+									MarkdownDescription: "The filter VQL for the cost metric.",
+								},
+							},
+							CustomType: resource_virtual_tag_config.CostMetricType{
+								ObjectType: types.ObjectType{
+									AttrTypes: resource_virtual_tag_config.CostMetricValue{}.AttributeTypes(ctx),
+								},
+							},
+							Optional: true,
```

Comment:
Somewhere I've made a mistake. `cost_metric` is optional, member fields are not.

However:
```
# ...
Must set a configuration value for the values[0].cost_metric.aggregation
attribute as the provider has marked it as required.
# ...
Must set a configuration value for the values[0].cost_metric.filter attribute
as the provider has marked it as required.
```

Reply:
no longer an issue post-relaxing the schema.

I suspect somewhere there's something like a bad default / zero value type mucking up the works but it was a real chore digging into things and re-orienting on the provider framework/our generated packages, so I'm hoping I can continue to punt true understanding or enlightenment just finds me.